src/Lecture3.hs

```diff
@@ -181,8 +218,8 @@ together only different elements.
 Product {getProduct = 6}
 
 -}
-appendDiff3 = error "TODO"
-
+appendDiff3 :: (Monoid a, Eq a) => a -> a -> a -> a
+appendDiff3 l1 l2 l3 = mconcat $ nub [l1, l2, l3]
```

Comment:
That's a clever and concise implementation! 👏🏻 

However, the type signature is more strict than required. Your function works with `Monoid`. But not every type can have a `Monoid` instance. This means, that `appendDiff3` can't work with types like `List1`. But if you change the constraint from `Monoid` to `Semigroup`, this function will now work. Though, relaxing the constraint will require changing the implementation as well 🚧

Reply:
Updated